server/resp/src/main/java/org/infinispan/server/resp/commands/json/JSONDEBUG.java

```diff
@@ -0,0 +1,79 @@
+package org.infinispan.server.resp.commands.json;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+import org.infinispan.server.resp.commands.connection.MemoryEntrySizeUtils;
+import org.infinispan.server.resp.json.EmbeddedJsonCache;
+import org.infinispan.server.resp.serialization.ResponseWriter;
+
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * JSON.DEBUG MEMORY
+ *
+ * @see <a href="https://redis.io/commands/json.debug-memory/">JSON.DEBUG MEMORY</a>
+ *
+ * @since 15.2
+ */
+public class JSONDEBUG extends RespCommand implements Resp3Command {
+
+   public JSONDEBUG() {
+      super("JSON.DEBUG", -2, 1, 1, 1);
+   }
+
+   protected JSONDEBUG(String name, int arity, int firstKeyPos, int lastKeyPos, int steps) {
+      super(name, arity, firstKeyPos, lastKeyPos, steps);
+   }
+
+   @Override
+   public long aclMask() {
+      return 0;
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler, ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+      String subcommand = new String(arguments.get(0)).toUpperCase();
+      if (subcommand.equals("HELP")) {
+         List<String> help = List.of(
+                 "MEMORY <key> [path] - reports memory usage",
+                 "HELP                - this message");
+         return handler.stageToReturn(CompletableFuture.completedFuture(help),
+                 ctx, ResponseWriter.ARRAY_STRING);
+      }
+
+      if (subcommand.equals("MEMORY")) {
+         byte[] key = arguments.get(1);
+         JSONCommandArgumentReader.CommandArgs commandArgs = JSONCommandArgumentReader.readCommandArgs(arguments, key, 2);
+         EmbeddedJsonCache ejc = handler.getJsonCache();
+         if (commandArgs.isRoot()) {
```

Comment:
I wonder, do we not want to just use a functional map function in the root case as well? This way we don't have to send the cache entry back across the wire and instead it is just the Long?

Reply:
I don't know to be honest. I believe since this command is called "debug" being a slow command is completely accepted, or sending the entry value. I really don't know if we are implementing this correctly

Reply:
Alright, I guess I am optimizing something that most likely is not meant to be performant anyways.